chat-server/server.js

```diff
@@ -21,8 +23,68 @@ const welcomeMessage = {
 //We will start with one message in the array.
 const messages = [welcomeMessage];
 
-app.get("/", (request, response) => {
-  response.sendFile(__dirname + "/index.html");
+app.get("/", (req, res) => {
+  res.sendFile(__dirname + "/index.html");
+});
+
+app.post("/messages", (req, res) => {
+  const newMessage = req.body;
+  if (
+    !newMessage.hasOwnProperty("text") ||
+    !newMessage.hasOwnProperty("from") ||
+    newMessage.from === "" ||
+    newMessage.text === "" ||
+    typeof newMessage.text != "string" ||
+    typeof newMessage.from != "string"
+  ) {
+    res.sendStatus(400).send("wrong input");
+  }
+  newMessage["id"] = messages[messages.length - 1].id + 1;
+  newMessage["timeSent"] = new Date();
+  messages.push(newMessage);
+  res.json(messages);
+});
+
+app.get("/messages", (req, res) => {
+  res.json(messages);
+});
+
+app.get("/messages/id/:id", (req, res) => {
+  const messageId = req.params.id;
+  const searchedMessage = messages.find((message) => message.id == messageId);
+  res.json(searchedMessage);
+});
+
+app.delete("/messages/id/:id", (req, res) => {
```

Comment:
Same as before!

Reply:
Can we set up a meeting for this issue? I couldn't find how to resolve dynamic and static path blocking